apifest-oauth20/src/main/java/com/apifest/oauth20/persistence/cassandra/CassandraDBManager.java

```diff
@@ -0,0 +1,510 @@
+            ResultSet rs = session.execute(stmt);
+            for (Row r : rs) {
+                AccessToken atoken = new AccessToken();
+                Row row = r;
+                mapRowToAccessToken(row, atoken);
+            }
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e);
+        }
+        return list;
+    }
+
+
+
+
+
+    @Override
+    public void storeAuthCode(AuthCode authCode) {
+        try {
+            Insert stmt = QueryBuilder.insertInto(KEYSPACE_NAME, AUTH_CODE_TABLE_NAME)
+                //.value("id", authCode.getId())
+                .value("code", authCode.getCode())
+                .value("client_id", authCode.getClientId())
+                .value("redirect_uri", authCode.getRedirectUri())
+                .value("state", authCode.getState())
+                .value("scope", authCode.getScope())
+                .value("type", authCode.getType())
+                .value("valid", authCode.isValid())
+                .value("user_id", authCode.getUserId())
+                .value("created", authCode.getCreated())
+            ;
+            session.execute(stmt);
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void updateAuthCodeValidStatus(String authCode, boolean valid) {
+        // 1. find 1 record by authCode
+        // 2. update valid flag
+        try {
+            Update.Where stmt = QueryBuilder.update(KEYSPACE_NAME, AUTH_CODE_TABLE_NAME)
+                .with(QueryBuilder.set("valid", valid))
+                .where(QueryBuilder.eq("code", authCode))
+            ;
+            session.execute(stmt);
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public AuthCode findAuthCode(String authCode, String redirectUri) {
+        try {
+            Select.Where stmt = QueryBuilder.select().from(KEYSPACE_NAME, AUTH_CODE_TABLE_NAME)
+                    .where(QueryBuilder.eq("code", authCode))
+                    .and(QueryBuilder.eq("redirect_uri", redirectUri));
+            //TODO: add valid = true condition
+            ResultSet rs = session.execute(stmt);
+            Iterator<Row> iter = rs.iterator();
+            if(iter.hasNext()) {
+                Row row = iter.next();
+                AuthCode ret = new AuthCode();
+                //ret.setId(row.getString("id"));
+                ret.setCode(row.getString("code"));
+                ret.setClientId(row.getString("client_id"));
+                ret.setRedirectUri(row.getString("redirect_uri"));
+                ret.setState(row.getString("state"));
+                ret.setScope(row.getString("scope"));
+                ret.setType(row.getString("type"));
+                ret.setValid(row.getBool("valid"));
+                ret.setUserId(row.getString("user_id"));
+                ret.setCreated(row.getTimestamp("created").getTime());
+                if(ret.isValid())
```

Comment:
Why not checking the validity in the beginning. If the code is not walid, there is no pount in filling out or even creating the "ret" structure.